pkg/utils.py

```diff
@@ -218,23 +215,38 @@ def process_test(
     logger.info("The test %s has started", test["name"])
     fingerprint_index = test["index"]
 
-    # getting metadata
-    metadata = extract_metadata_from_test(test) if options["uuid"] in ("", None) else get_metadata_with_uuid(options["uuid"], match)
-    # get uuids, buildUrls matching with the metadata
-    runs = match.get_uuid_by_metadata(metadata, fingerprint_index, lookback_date=start_timestamp, lookback_size=options['lookback_size'])
-    uuids = [run["uuid"] for run in runs]
-    buildUrls = {run["uuid"]: run["buildUrl"] for run in runs}
-    # get uuids if there is a baseline
+    # get uuids if there is a baseline and uuid set
     if options["baseline"] not in ("", None):
+        # if baseline is set, set uuids
         uuids = [uuid for uuid in re.split(r" |,", options["baseline"]) if uuid]
         uuids.append(options["uuid"])
-        buildUrls = get_build_urls(fingerprint_index, uuids, match)
-    elif not uuids:
+        runs = match.getResults("", uuids, fingerprint_index, {})
+        # get metadata of one run
+        metadata = get_metadata_with_uuid(options["uuid"], match)
+    else:
+        # getting metadata
+        metadata = extract_metadata_from_test(test) if options["uuid"] in ("", None) else get_metadata_with_uuid(options["uuid"], match)
+        # get uuids, buildUrls matching with the metadata
+        # this match might not always work if UUID failed run, we still want to analyze
+        runs = match.get_uuid_by_metadata(metadata, fingerprint_index, lookback_date=start_timestamp, lookback_size=options['lookback_size'])
+
+        if options['previous_version']:
+            last_version_run = runs
+            metadata['ocpVersion'] = str(float(metadata['ocpVersion'][:4]) - .01)
+            runs = match.get_uuid_by_metadata(metadata, fingerprint_index, lookback_date=start_timestamp, lookback_size=options['lookback_size'])
+            if len(last_version_run) > 0:
+                # get latest uuid as the "uuid" to compare against
+                last_uuid_run = last_version_run[-1]
```

Comment:
thinking here might be better to get the last (0) of the specified version. If not, when giving the 15 day lookback it'll select the oldest UUID. Thoughts?